providers/oidc.go

```diff
@@ -39,10 +39,6 @@ func NewOIDCProvider(p *ProviderData, opts options.OIDCOptions) *OIDCProvider {
 		scope:       oidcDefaultScope,
 	}
 
-	if len(p.AllowedGroups) > 0 {
-		oidcProviderDefaults.scope += " groups"
-	}
```

Comment:
this will be breaking change for people who rely on this implicit behaviour.

therefore we should introduce a flag to deactivate it rather than removing it.